generator/json-generator.lua

```diff
@@ -1,135 +1,151 @@
+    if vdef.arguments then
+        variant.args = createArguments( vdef.arguments )
+    end
+
+    if vdef.returns then
+        variant.returnTypes = createReturnTypes( vdef.returns )
+    end
+    return variant
 end
 
-local function buildModuleFunctionCompletion( f, module, closing )
-    local arguments = generateArguments( f.variants[1].arguments )
-    return {
-        TAB .. '{' .. LINE_BREAK,
-        TAB .. TAB .. '"displayText": "love.' .. module.name .. '.' .. f.name .. APOSTROPHE .. COMMA .. LINE_BREAK,
-        TAB .. TAB .. '"type": "function"' .. COMMA .. LINE_BREAK,
-        TAB .. TAB .. '"description": ' .. APOSTROPHE .. cleanUpString( f.description ) .. APOSTROPHE .. COMMA .. LINE_BREAK,
-        TAB .. TAB .. '"descriptionMoreURL": ' .. APOSTROPHE .. WIKI_URL .. string.format( 'love.%s.%s', module.name, f.name ) .. APOSTROPHE .. COMMA .. LINE_BREAK,
-        TAB .. TAB .. '"snippet": ',
-        APOSTROPHE .. string.format( 'love.%s.%s(%s)', module.name, f.name, arguments ) .. APOSTROPHE .. LINE_BREAK,
-        TAB .. (  closing and '}' or '},'  ) .. LINE_BREAK
-    }
+local function createFunction( f, parent, moduleString )
+    local name = f.name
+    parent[name] = {}
+    parent[name].type = 'function'
+
+    parent[name].description = f.description
+    parent[name].link = string.format( "%s%s%s", WIKI_URL, moduleString, name )
+
+    -- Create normal function if there is just one variant.
+    if #f.variants == 1 then
+        local v = createVariant( f.variants[1] )
+        parent[name].args = v.args
+        parent[name].returnTypes = v.returnTypes
+        return
+    end
+
+    -- Generate multiple variants.
+    parent[name].variants = {}
+    for i, v in ipairs( f.variants ) do
+        parent[name].variants[i] = createVariant( v )
+    end
 end
 
-local function buildTypeFunctionCompletion( f, type, closing )
-    local arguments = generateArguments( f.variants[1].arguments, 2 )
-    return {
-        TAB .. '{' .. LINE_BREAK,
-        TAB .. TAB .. '"displayText": "' .. type.name .. ':' .. f.name .. APOSTROPHE .. COMMA .. LINE_BREAK,
-        TAB .. TAB .. '"type": "function"' .. COMMA .. LINE_BREAK,
-        TAB .. TAB .. '"description": ' .. APOSTROPHE .. cleanUpString( f.description ) .. APOSTROPHE .. COMMA .. LINE_BREAK,
-        TAB .. TAB .. '"descriptionMoreURL": ' .. APOSTROPHE .. WIKI_URL .. string.format( '%s:%s', type.name, f.name ) .. APOSTROPHE .. COMMA .. LINE_BREAK,
-        TAB .. TAB .. '"snippet": ',
-        APOSTROPHE .. string.format( '${%d:%s}:%s(%s)', 1, type.name, f.name, arguments ) .. APOSTROPHE .. LINE_BREAK,
-        TAB .. (  closing and '}' or '},'  ) .. LINE_BREAK
-    }
+local function createLoveFunctions( functions, parent )
+    for _, f in pairs( functions ) do
+        createFunction( f, parent, LOVE_MODULE_STRING )
+    end
 end
 
-local function createJSON()
-    print( 'Generating LOVE snippets ... ' )
+local function createLoveCallbackFunctions( callbacks, parent )
```

Comment:
Isn't this function THE SAME as `createLoveFunctions`? I'm if you want to keep it to make the code easier to read